### ISSUE TITLE
SAPT Jobs

### ISSUE THREAD

**Issue by polikw** [psi4/psi4webmo#15]
SAPT fails, as I naively try to run it.  How do I create a valid input file from within WebMO? (job 9: CHFO - fail)

[9.zip](https://github.com/psi4/psi4webmo/files/294241/9.zip)

**Comment by loriab** [comment]
70e73c68a8ae20190d699482b303ea2dde4f1156 may help this problem.

**Comment by polikw** [comment]
Could someone post a very simple SAPT file that runs, or run a simple SAPT job on
    http://104.197.158.156/~webmo/cgi-bin/webmo/login.cgi
(username = student, password = student)

Thanks!

**Comment by loriab** [comment]
Job 25 on that server is a SAPT job that runs fine. I suspect what's happening in your example is that the CO and HF fragments are too near to be separated by the breadth first search algorithm, so the autofrag is listing them followed by a `--` fragment separator, which the molecule string parser doesn't take kindly to at the end of a molecule. The other job I tried failed because `ssapt0` isn't a proper argument to `energy()`. I really thought it was. Anyway, removed it (ed7e224c6b1786e05b0182ed91d0ad6ff5ff75d4)  as per this and #20 . So SAPT really does work, I believe, if you run sapt0 on a suitable mol.

**Comment by polikw** [comment]
Although SAPT job 25 might "run fine", it does not parse properly.  In particular, the file "output.xyz" that is supposed to contain the final geometry is empty.  You also notice this if you view the job results from within WebMO, at which point no structure is displayed.

In order to support SAPT, parse_psi.pl needs a tiny bit of work in order to be able to find the final geometry (which I see is in the output file at the end).

Also, you might want to parse out some of the more interesting SAPT-related values from the variable map at the end, e.g., SAPT ENERGY.  Personally, I am not sure which of these are most useful, but you probably want WebMO to display those quantities you feel are most useful.

In summary, it is worth visiting Job 25 at
    http://104.197.158.156/~webmo/cgi-bin/webmo/login.cgi
    (username = student, password = student)
looking at the raw output, deciding how one could parse the final geometry, and editing parse_psi.pl accordingly.  Optionally, you may want to parse out some of the SAPT-related values in the Variable map at the end.

**Comment by loriab** [comment]
Aha, so that's why the molecule disappeared at the end of the SAPT job. It looks like the WebMO output is catching Elst, Exch, Ind, Disp. So @bwb314 is going to add another line for total SAPT interaction energy in kcal/mol (which is the other important bit from the Variable Map) and fix the final geom (which is the same as the input geom).

Whoops, sorry on the premature #20 fix. I'll leave the psi4 WebMO code editing to the psi4 WebMO users. :-)

**Comment by bwb314** [comment]
Commit 4e6c566 fixes geometry parsing for SAPT jobs and parses out the SAPT interaction energy as @loriab suggested.

**Comment by polikw** [comment]
I concur and have verified that SAPT job geometries are now parsed.  (Just a comment:  I presume that you are parsing the "final" geometry, even if the initial and final geometry are the same.  That is what WebMO typically does.)

Also, Total Interaction Energy is now parsed.

However, there might be a minor issue for psi4 regarding labeling of values at the end.  Specifically, the SCF energy is reported as being -76 Hartree.  But that's the energy of a single H2O molecule.  I believe that the SCF Energy is just that of the final monomer, not of the entire system.  Some additional clarity in the psi4 variable names would be useful for a novice user.

FYI, the complete list of variables (for H2O dimer, job 29) by WebMO is:

SCF Energy  -76.037943834749 Hartree
Dispersion Energy   -2.15867719928899 kcal/mol
Electrostatics Energy   -12.4527753505464 kcal/mol
Exchange Energy 16.9630467906415 kcal/mol
Induction Energy    -1.99469995590257 kcal/mol
Total Interaction Energy    0.356894284903609 kcal/mol
Dipole Moment   2.12951756334387 Debye

and the variable map as reported by psi4 is:

  Variable Map:

---

  "CURRENT DIPOLE X"            =>       1.965938591233
  "CURRENT DIPOLE Y"            =>       0.809219716390
  "CURRENT DIPOLE Z"            =>       0.122857473100
  "CURRENT ENERGY"              =>       0.000568747222
  "CURRENT REFERENCE ENERGY"    =>     -76.037943834749
  "HF TOTAL ENERGY"             =>     -76.037943834749
  "NUCLEAR REPULSION ENERGY"    =>       9.353552289498
  "ONE-ELECTRON ENERGY"         =>    -123.243580819658
  "SAPT CT ENERGY"              =>      -0.001788389707
  "SAPT DISP ENERGY"            =>      -0.003440070946
  "SAPT DISP20 ENERGY"          =>      -0.004784792410
  "SAPT DISP20(OS) ENERGY"      =>      -0.002392396205
  "SAPT DISP20(SS) ENERGY"      =>      -0.002392396205
  "SAPT ELST ENERGY"            =>      -0.019844759881
  "SAPT ELST10,R ENERGY"        =>      -0.019844759881
  "SAPT ENERGY"                 =>       0.000568747222
  "SAPT EXCH ENERGY"            =>       0.027032334635
  "SAPT EXCH-DISP20 ENERGY"     =>       0.001344721464
  "SAPT EXCH-DISP20(OS) ENERGY" =>       0.000623036039
  "SAPT EXCH-DISP20(SS) ENERGY" =>       0.000721685425
  "SAPT EXCH-IND20,R ENERGY"    =>       0.007744446408
  "SAPT EXCH10 ENERGY"          =>       0.027032334635
  "SAPT EXCH10(S^2) ENERGY"     =>       0.026700066570
  "SAPT EXCHSCAL"               =>       1.012444465779
  "SAPT EXCHSCAL3"              =>       1.037799918730
  "SAPT HF TOTAL ENERGY"        =>       0.004008818168
  "SAPT HF(2) ALPHA=0.0 ENERGY" =>      -0.001390366879
  "SAPT HF(2) ENERGY"           =>      -0.001486742378
  "SAPT IND ENERGY"             =>      -0.003178756586
  "SAPT IND20,R ENERGY"         =>      -0.009532836115
  "SAPT SAPT0 ENERGY"           =>       0.000568747222
  "SAPT0 DISP ENERGY"           =>      -0.003423336606
  "SAPT0 ELST ENERGY"           =>      -0.019844759881
  "SAPT0 EXCH ENERGY"           =>       0.027032334635
  "SAPT0 INDC ENERGY"           =>      -0.003178756586
  "SAPT0 TOTAL ENERGY"          =>       0.000585481562
  "SAPT2 DISP ENERGY"           =>      -0.003423336606
  "SCF DIPOLE X"                =>       1.965938591233
  "SCF DIPOLE Y"                =>       0.809219716390
  "SCF DIPOLE Z"                =>       0.122857473100
  "SCF ITERATION ENERGY"        =>     -76.037943834749
  "SCF TOTAL ENERGY"            =>     -76.037943834749
  "SCS-SAPT0 DISP ENERGY"       =>      -0.003225901314
  "SCS-SAPT0 ELST ENERGY"       =>      -0.019844759881
  "SCS-SAPT0 EXCH ENERGY"       =>       0.027032334635
  "SCS-SAPT0 INDC ENERGY"       =>      -0.003178756586
  "SCS-SAPT0 TOTAL ENERGY"      =>       0.000782916854
  "SSAPT0 DISP ENERGY"          =>      -0.003389240584
  "SSAPT0 ELST ENERGY"          =>      -0.019844759881
  "SSAPT0 EXCH ENERGY"          =>       0.027032334635
  "SSAPT0 INDC ENERGY"          =>      -0.002886017141
  "SSAPT0 TOTAL ENERGY"         =>       0.000912317028
  "TWO-ELECTRON ENERGY"         =>      37.852084695411
